### PR TITLE
nsiqunittest/nsiqcppstyle_unittestbase: Improve test readability

### PR DESCRIPTION
**Pull request by Farigh**
Replaced CheckErrorContent method with two much more readable ones:
  - ExpectError
  - ExpectSuccess

### CODE REVIEW

**Comment by Farigh** [comment]
Calling `self.ExpectError` seems more readable than `assert CheckErrorContent`
And `self.ExpectSuccess` seems more readable than `assert not CheckErrorContent`

Plus, the `ExpectError` and `ExpectSuccess` uses the python `assertTrue` and `assertFalse` methods

The output has been improved as well on error.
Before : 
```
======================================================================
FAIL: test1 (RULE_6_5_B_do_not_use_macro_for_constants.testRule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "F:\git_repo\github\nsiqcppstyle\rules\RULE_6_5_B_do_not_use_macro_for_constants.py", line 51, in test1
    assert not CheckErrorContent(__name__)
AssertionError

======================================================================
FAIL: test2 (RULE_6_5_B_do_not_use_macro_for_constants.testRule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "F:\git_repo\github\nsiqcppstyle\rules\RULE_6_5_B_do_not_use_macro_for_constants.py", line 56, in test2
    assert CheckErrorContent(__name__)
AssertionError

----------------------------------------------------------------------
```

After :
```
======================================================================
FAIL: test1 (RULE_6_5_B_do_not_use_macro_for_constants.testRule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "F:\git_repo\github\nsiqcppstyle\rules\RULE_6_5_B_do_not_use_macro_for_constants.py", line 51, in test1
    self.ExpectSuccess(__name__)
  File "nsiqunittest\nsiqcppstyle_unittestbase.py", line 61, in ExpectSuccess
    self.assertFalse(result, "Expected no error but got: " + str(errors))
AssertionError: Expected no error but got: [(LexToken(ID,'k',2,9,9, False, True), 'RULE_6_5_B_do_not_use_macro_for_constants', 'Do not use macro(k) for constant')]

======================================================================
FAIL: test2 (RULE_6_5_B_do_not_use_macro_for_constants.testRule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "F:\git_repo\github\nsiqcppstyle\rules\RULE_6_5_B_do_not_use_macro_for_constants.py", line 57, in test2
    self.ExpectError(__name__)
  File "nsiqunittest\nsiqcppstyle_unittestbase.py", line 56, in ExpectError
    self.assertTrue(result, "Expected error but got none")
AssertionError: Expected error but got none

----------------------------------------------------------------------
```

File: rules/RULE_4_5_A_braces_for_function_definition_should_be_located_in_seperate_line.py [hunk 1]
```diff
@@ -90,11 +94,13 @@ class A {
...
+        self.ExpectSuccess(__name__)
+
     def test5(self):
         self.Analyze("thisfile.c","""
 class K {
     void function() const
     {   }
 }
 """)
+        self.ExpectSuccess(__name__)
```
Comment (kunaltyagi) [inline rules/RULE_4_5_A_braces_for_function_definition_should_be_located_in_seperate_line.py]: Good catch!!

File: nsiqunittest/nsiqcppstyle_unittestbase.py [hunk 2]
```diff
@@ -33,17 +33,9 @@
 import nsiqcppstyle_state
 
 errors = []
-def AddError(err):
-    errors.append(err)
-
-def CheckErrorContent(msg):
```
[the remark this answers is not in the document]
Comment (Farigh) [reply nsiqunittest/nsiqcppstyle_unittestbase.py]: I moved it within the class since `ExpectError` and `ExpectSuccess` call are the only ones

**Comment by Farigh** [comment]
Done, thanks for your review

**Comment by kunaltyagi** [comment]
As an aside, might I ask you why, how and where are you using this library?

**Comment by Farigh** [comment]
I'm currently using on a C++ project to complement cpp-check with some custom coding-style rules.
Both being executed at compile time

**Pull request merged**